source/h2_decoder.c

```diff
@@ -314,10 +323,16 @@ struct aws_h2_decoder *aws_h2_decoder_new(struct aws_h2_decoder_params *params)
         goto array_list_failed;
     }
 
+    if (aws_byte_buf_init(
+            &decoder->header_block_in_progress.cookies, decoder->alloc, s_decoder_cookie_buffer_initial_size)) {
+        goto buffer_init_failed;
+    }
+
     return decoder;
 
-failed_new_hpack:
+buffer_init_failed:
 array_list_failed:
+failed_new_hpack:
     aws_mem_release(params->alloc, allocation);
 failed_alloc:
     return NULL;
```

Comment:
if we jump to `array_list_failed:` or `buffer_init_failed:` then there's stuff we fail to clean up.
maybe make this less error-prone by just having one `error:` label, like:
```
error:
if (decoder) {
  aws_hpack_context_destroy(...)
  aws_array_list_clean_up(...)
  aws_byte_buf_clean_up(...)
}
aws_mem_release(params->alloc, allocation)
```

We've tried to make all of our clean_up()/release()/destroy() functions safe to call on something whose initialization failed, or never happened (as long as we zeroed it all out to start)